### PR TITLE
[janitor] Refactor program logic

### PR DESCRIPTION
**Pull request by phannebohm**
Also:
- ~improve List.appendElt~
- fix typos

### CODE REVIEW

File: OMCompiler/Compiler/Util/List.mo [hunk 1]
```diff
@@ -478,7 +478,7 @@ public function appendElt<T>
   input list<T> inList;
   output list<T> outList;
 algorithm
-  outList := listAppend(inList, {inElement});
+  outList := listReverse(inElement :: listReverse(inList));
```
Comment (phannebohm) [inline OMCompiler/Compiler/Util/List.mo:481]: @perost we think `listReverse` is linear and `listAppend` is quadratic, is that correct?

Comment (perost) [reply OMCompiler/Compiler/Util/List.mo:481]: Why would `listAppend` be quadratic? It just allocates a new list that's large enough to contain both the lists. A `listAppend` is probably a bit more expensive compared with a `listReverse`, but I'm not sure about one `listAppend` vs two `listReverse`. You'll have to benchmark that to check what's faster.

Comment (phannebohm) [reply OMCompiler/Compiler/Util/List.mo:481]: I see. We thought `listAppend` would cons the left list element by element. But in that case I'll revert this.

Comment (perost) [reply OMCompiler/Compiler/Util/List.mo]: That's kind of what happens, but cons has constant complexity so cons:ing a list is still just linear. Here's the implementation of `listAppend` for reference:

https://github.com/OpenModelica/OpenModelica/blob/c94e2ebbd97fbe9b3bed444316d85702b4718338/OMCompiler/SimulationRuntime/c/meta/meta_modelica_builtin.c#L440-L464

**Review by kabdelhak — COMMENTED** [review]
LGTM

**Pull request merged**